diode-server/proto/reconciler/v1/reconciler.proto

```diff
@@ -36,8 +37,67 @@ message RetrieveIngestionDataSourcesResponse {
   repeated IngestionDataSource ingestion_data_sources = 1;
 }
 
+// ChangeSetError represents an error when applying a change set
+message ChangeSetError {
+  message Details {
+    string change_set_id = 1;
+    string result = 2;
+    message Error {
+      string error = 1; //key value pair of the error
+      string change_id = 2;
+    }
+    repeated Error errors = 3;
+  }
+  string message = 1;
+  int32 code = 2;
+  Details details = 3;
+}
+
+enum State {
```

Comment:
@mfiedorowicz what do you think about this?

the proto generated code is:


```
type State int32

const (
	State_NEW        State = 0
	State_RECONCILED State = 1
	State_FAILED     State = 2
	State_NO_CHANGES State = 3
)

// Enum value maps for State.
var (
	State_name = map[int32]string{
		0: "NEW",
		1: "RECONCILED",
		2: "FAILED",
		3: "NO_CHANGES",
	}
	State_value = map[string]int32{
		"NEW":        0,
		"RECONCILED": 1,
		"FAILED":     2,
		"NO_CHANGES": 3,
	}
)

func (x State) Enum() *State {
	p := new(State)
	*p = x
	return p
}
```

Reply:
Looks good. There is general recommendation to add `*_UNSPECIFIED = 0` first enum value (https://protobuf.dev/programming-guides/dos-donts/#unspecified-enum), but in our case it's for output (ingestion log) rather than input, so it's fine IMHO.

Reply:
in our case I don't think it makes sense and also when filtering it is optional value 0 is NEW for us